Sources/Danger/Plugins/SwiftLint/SwiftLint.swift

```diff
@@ -46,86 +46,115 @@ extension SwiftLint {
-            violations = makeViolations(from: outputJSON, failAction: failAction)
+        guard !violations.isEmpty else {
+            return []
         }
 
         let currentPath = currentPathProvider.currentPath
         violations = violations.map { violation in
-            let updatedPath = violation.file.deletingPrefix(currentPath).deletingPrefix("/")
             var violation = violation
+
+            let updatedPath = violation.file.deletingPrefix(currentPath).deletingPrefix("/")
             violation.file = updatedPath
+
             if strict {
                 violation.severity = .error
             }
             return violation
         }
 
-        if !violations.isEmpty {
-            if inline {
-                violations.forEach { violation in
-                    switch violation.severity {
-                    case .error:
-                        failInlineAction(violation.messageText, violation.file, violation.line)
-                    case .warning:
-                        warnInlineAction(violation.messageText, violation.file, violation.line)
-                    }
+        if inline {
+            violations.forEach { violation in
+                switch violation.severity {
+                case .error:
+                    failInlineAction(violation.messageText, violation.file, violation.line)
+                case .warning:
+                    warnInlineAction(violation.messageText, violation.file, violation.line)
                 }
-            } else {
-                var markdownMessage = """
-                ### SwiftLint found issues
-
-                | Severity | File | Reason |
-                | -------- | ---- | ------ |\n
-                """
-                markdownMessage += violations.map { $0.toMarkdown() }.joined(separator: "\n")
-                markdownAction(markdownMessage)
             }
+        } else {
+            var markdownMessage = """
+            ### SwiftLint found issues
+
+            | Severity | File | Reason |
+            | -------- | ---- | ------ |\n
+            """
+            markdownMessage += violations.map { $0.toMarkdown() }.joined(separator: "\n")
+            markdownAction(markdownMessage)
         }
 
         return violations
     }
 
+    private static func lintAll(directory: String?,
+                                arguments: inout [String],
+                                shellExecutor: ShellExecutor,
+                                swiftlintPath: String,
+                                failAction: (String) -> Void) -> [SwiftLintViolation] {
+        if let directory = directory {
+            arguments.append("--path \"\(directory)\"")
+        }
+
+        let outputJSON = shellExecutor.execute(swiftlintPath, arguments: arguments)
+        return makeViolations(from: outputJSON, failAction: failAction)
+    }
+
+    private static func lintModifiedAndCreated(danger: DangerDSL,
```

Comment:
<!--
  0 failure: 
  1 warning:  Function should h...
  
  
  DangerID: danger-id-default;
  File: Sources/Danger/Plugins/SwiftLint/SwiftLint.swift;
  Line: 125;
-->

- :warning: Function should have 5 parameters or less: it currently has 6 (`function_parameter_count`)